docs/manage-jobs-in-gravitino.md

```diff
@@ -0,0 +1,583 @@
+      .customFields(Map.of("field1", "{{value1}}", "field2", "{{value2}}"))
+      .scripts(List.of("/path/to/script1.sh", "/path/to/script2.sh"))
+      .build();
+  
+  GravitinoClient client = ...;
+  client.registerJobTemplate(jobTemplate);
+```
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+```python
+  shell_job_template = (
+      ShellJobTemplate.builder()
+      .with_name("my_shell_job_template")
+      .with_comment("A shell job template to run a script")
+      .with_executable("/path/to/my_script.sh")
+      .with_arguments(["{{arg1}}", "{{arg2}}"])
+      .with_environments({"ENV_VAR1": "{{value1}}", "ENV_VAR2": "{{value2}}"})
+      .with_custom_fields({"field1": "{{value1}}", "field2": "{{value2}}"})
+      .with_scripts(["/path/to/script1.sh", "/path/to/script2.sh"])
+      .build()
+  )
+
+  client = GravitinoClient(...)
+  client.register_job_template(shell_job_template)
+```
+
+</TabItem>
+</Tabs>
+
+### List registered job templates
+
+You can list all the registered job templates under a metalake by using the REST API or the Java
+and Python SDKs.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X GET -H "Accept: application/vnd.gravitino.v1+json" \
+     http://localhost:8090/api/metalakes/test/jobs/templates
+     
+Or using query parameter "details=true" to get more details of the job templates:
+
+curl -X GET -H "Accept: application/vnd.gravitino.v1+json" \
+     http://localhost:8090/api/metalakes/test/jobs/templates?details=true
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+    GravitinoClient client = ...;
+    List<JobTemplate> detailedJobTemplates = client.listJobTemplates();
+```
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+```python
+    client = GravitinoClient(...)
+    detailed_job_templates = client.list_job_templates()
+```
+
+</TabItem>
+</Tabs>
+
+### Get a registered job template by name
+
+You can get a registered job template by its name using the REST API or the Java and Python SDKs.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X GET -H "Accept: application/vnd.gravitino.v1+json" \
+     http://localhost:8090/api/metalakes/test/jobs/templates/my_shell_job_template
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+    GravitinoClient client = ...;
+    JobTemplate jobTemplate = client.getJobTemplate("my_shell_job_template");
+```
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+```python
+    client = GravitinoClient(...)
+    job_template = client.get_job_template("my_shell_job_template")
+```
+
+</TabItem>
+</Tabs>
+
+### Delete a registered job template by name
+
+You can delete a registered job template by its name using the REST API or the Java and Python SDKs.
+
+Note that deleting a job template will also delete all the jobs that are using this job template.
```

Comment:
Will we delete all completed (FAILED, SUCCEEDED, CANCELED) jobs that used this job template?

Reply:
Yes, all the jobs related to this template will be removed. Because the template is not existed anymore, if we can still list the jobs with non-existed job template, it will be misleading.